### PR TITLE
Add Comprehensive Free Resources for Algorithms and System Design.

### PR DESCRIPTION
**Pull request by shreykumar33**
What does this PR do?
Add resource(s)

For resources
Description
I added some courses on algorithms and system design.

Why is this valuable (or not)?
The courses are really valuable.

How do we know it's really free?
Links to the playlist of courses provided are globally accessible.

For book lists, is it a book? For course lists, is it a course? etc.
Yes, courses contain a proper roadmap with video lectures.

Checklist:
 Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
 [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
 Include author(s) and platform where appropriate.
 Put lists in alphabetical order, correct spacing.
 Add needed indications (PDF, access notes, under construction).
 Used an informative name for this pull request.

### CODE REVIEW

**Review by shreykumar33 — COMMENTED** [review]
fixed lines

**Comment by shreykumar33** [comment]
> Linter failed, fix the error(s):
> 
> ```
> free-programming-books/courses/free-courses-ur.md
> 65:1-66:3  warning  Incorrect number of blank lines between last section and next heading  blank-lines-1-0-2  remark-lint
> ```

Fixed

**Comment by shreykumar33** [comment]
> Linter failed, fix the error(s):
> 
> ```
> free-programming-books/courses/free-courses-ur.md
> 65:1-66:3  warning  Incorrect number of blank lines between last section and next heading  blank-lines-1-0-2  remark-lint
> ```

Fixed

**Review by shreykumar33 — COMMENTED** [review]
All resources checked and issues fixed. Kindly review

File: courses/free-courses-ur.md [hunk 1]
```diff
@@ -9,11 +9,14 @@
 * [HTML and CSS](#html-and-css)
 * [Java](#java)
 * [Python](#python)
+* [System Design](#system-design)
 
 
 ### Algorithms
 
+* [Complete C++ Placement DSA Course](https://www.youtube.com/watch?v=WQoB2z67hvY&list=PLDzeHZWIZsTryvtXdMr6rPh4IDexB5NIA) - Love Babbar
```
Comment (eshellman) [inline courses/free-courses-ur.md:17]: For youtube, we prefer "playlist" links and the name of the producer/creator. docs/CONTRIBUTING.md#platforms-and-access-notes

File: courses/free-courses-ur.md [hunk 2]
```diff
@@ -9,11 +9,14 @@
...
 * [Data Structures and Algorithms Analysis Complete Course in Hindi/Urdu \| Data structures by Fahad Hussain \| data structures and algorithms tutorial](https://www.youtube.com/playlist?list=PLtCBuHKmdxOfPNlAKWxBqdznCcXV4iWCz) - Fahad Hussain
+* [Striver's DSA playlist](https://takeuforward.org/array/top-array-interview-questions-structured-path-with-video-solutions/) - Striver
```
Comment (eshellman) [inline courses/free-courses-ur.md:19]: not urdu

File: courses/free-courses-ur.md [hunk 3]
```diff
@@ -60,4 +63,11 @@
 ### Python
 
 * [Python_ka_chilla (python in 40 days in urdu/Hindi)](https://www.youtube.com/playlist?list=PL9XvIvvVL50HVsu-Ao8NBr0UJSO8O6lBI) - Codeanics
-  
+
+
+### System Design
+
+* [System Design Roadmap with Videos for SDEs](https://takeuforward.org/system-design/complete-system-design-roadmap-with-videos-for-sdes/) - Striver
```
Comment (eshellman) [inline courses/free-courses-ur.md:70]: not urdu

**Review by eshellman — CHANGES_REQUESTED** [review]

**Comment by eshellman** [comment]
all our other takeuforward items are in the problem sets/competitive programming list, including items that seem very close to these. Please consider using that list instead of the courses list.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**